samples/sqltransport/startup-purge-behavior/SqlTransport_6/Shared/SqlHelper.cs

```diff
@@ -0,0 +1,60 @@
+﻿using Microsoft.Data.SqlClient;
+
+public static class SqlHelper
+{
+    public static void ExecuteSql(string connectionString, string sql)
+    {
+        EnsureDatabaseExists(connectionString);
+
+        using (var connection = new SqlConnection(connectionString))
+        {
+            connection.Open();
+
+            using (var command = connection.CreateCommand())
+            {
+                command.CommandText = sql;
+                command.ExecuteNonQuery();
+            }
+        }
+    }
+
+    public static void CreateSchema(string connectionString, string schema)
+    {
+        var sql = $@"
+if not exists (select  *
+               from    sys.schemas
+               where   name = N'{schema}')
+    exec('create schema {schema}');";
+        ExecuteSql(connectionString, sql);
+    }
+
+    public static void EnsureDatabaseExists(string connectionString)
+    {
+        var builder = new SqlConnectionStringBuilder(connectionString);
+        var database = builder.InitialCatalog;
+
+        var masterConnection = connectionString.Replace(builder.InitialCatalog, "master");
+
+        using (var connection = new SqlConnection(masterConnection))
+        {
+            connection.Open();
+
+            using (var command = connection.CreateCommand())
+            {
+                command.CommandText = $@"
+if(db_id('{database}') is null)
+    create database [{database}]
+";
+                command.ExecuteNonQuery();
+            }
+        }
+    }
+
+    public static void TruncateMessageTable(string connectionString, string endpointName)
+    {
+        ExecuteSql(connectionString, $@"IF EXISTS (SELECT TOP 1 1 FROM sys.objects WHERE name = 'Samples.SqlServer.TruncateReceiver')
```

Comment:
Would it make sense to introduce a parameter for the main input queue. We could also show how to get a transport input queue name based on the endpoint name - this would handle non-default schema and catalog configuration. The only way to do it that I know if is to use `TransportInfrastructure.ToTransportAddress` that is registered by the transport seam in `Settings`. Here is an [example](https://github.com/Particular/NServiceBus/blob/master/src/NServiceBus.Core/Routing/SubscriptionMigrationMode/SubscriptionMigrationMode.cs#L43) of this being used in Core.

Reply:
Done